backend/src/impl/default_controllers_impl.py

```diff
@@ -220,7 +237,27 @@ def benchmark_get_by_id(benchmark_id: str, by_creator: bool) -> Benchmark:
 
 
 def systems_get_by_id(system_id: str) -> System:
-    return SystemDBUtils.find_system_by_id(system_id)
+    system = SystemDBUtils.find_system_by_id(system_id)
+    user = get_user()
+    if system.is_private:
+        if user.is_authenticated:
+            if _is_creator(system, user):
+                return system
+            elif _is_shared_user(system, user):
+                # must be creator to see shared_users
+                system.shared_users = None
```

Comment:
Do we need to hide this information from the shared users? I think this adds some complications that may not be necessary. It'll also be great if we can make sure the behavior is consistent for **GET /systems** and **GET /systems/{id}**. I think **GET /systems** doesn't have this logic implemented at the moment.

Our access control model is simple so we haven't really made this concrete. But I think our current model basically looks something like this:

- We have two permission types: Read (R) and Write (W)
- creator: R + W
- shared user: R
- for public systems, any user: R

If we restrict shared users from reading `shared_users`, you are adding a third permission type. It is also an ad hoc permission type that says read everything else but not `shared_users`. If we create more types like this, I think it makes the access control system hard to manage.

Reply:
Thanks! I was hesitant about shared users having access to each other's emails, but perhaps I guess as the first step and in favor of simplicitly, we should go with the access control model you showed.
Now both `GET /systems/{id}` and `GET /systems`s' behaviors match the 2-permission-type model.